### ISSUE TITLE
Add `--no-daemon` or `--foreground` option for supervised systems

### ISSUE THREAD

**Issue by jac-cbi** [cloudflare/cloudflared#315]
I've recently moved most of my servers and VMs to Artix Linux (a derivative of Arch Linux, sans systemd).  I chose the `runit` system for my init and process supervision system on my Artix boxes (s6 is a mess, and OpenRC, well, it's not a supervision system).  This has been working great up until I attempted to move my web server from Gentoo to Artix.  Everything went smoothly, except for `cloudflared`.

I use `cloudflared` for connecting to the Cloudflare network, which caches, and serves my websites.  It's a traditional use of Argo tunnel, namely, my web server is *not* accessible from the Internet, only from CF via Argo tunnel. 

Basically, If CF were to add a `--no-daemon` option, this would allow me to launch the process in a supervised manner, meaning that there is a parent process which receives SIGCHLD and can automatically restart the dead process.  Additionally, this means I could allow `cloudflared` to automagically update itself, and once complete, it can just exit.  No races, no PID tracking, and more importantly, no dropped log messages. (runit handles that).

**Comment by stevenanthonyrevo** [comment]
Upvote for including this, PID tracking is a pain.

**Comment by andrewheberle** [comment]
I must be missing something here...but `cloudflared` has always run in the foreground and runs fine under a supervision system in my experience...
